internal/pkg/aws/s3/s3.go

```diff
@@ -52,3 +61,64 @@ func (s *Service) PutArtifact(bucket, fileName string, data io.Reader) (string,
 
 	return resp.Location, nil
 }
+
+// EmptyBucket deletes all objects within the bucket.
+func (s *Service) EmptyBucket(bucket string) error {
```

Comment:
Check this out! https://github.com/awsdocs/aws-doc-sdk-examples/blob/bc948130e67d8f9a3adb8c1980ddf5405aed4e33/go/example_code/s3/s3_delete_objects.go#L62-L69

I think using the `BatchDeleteIterator` can make our lives a lot easier :), let me know if I'm missing something

Reply:
Yeah, problem with this method is we are using versioning bucket so this cannot entirely empty the bucket. Instead this will create delete marker for each file for the next version. And the reason why we need to delete "delete marker" at the end of this function is to deal with the edge case where users might delete some of the files themselves so that "delete marker" will be created, which will not be deleted by the code snippet in the "for loop".